portfolio/src/main/java/com/google/sps/objects/User.java

```diff
@@ -0,0 +1,42 @@
+package com.google.sps.objects;
+
+import java.util.Objects;
+
+public final class User {
+  private final String userName;
+  private final String commentBody;
+
+  public User(String userName, String commentBody) {
+    this.userName = userName;
+    this.commentBody = commentBody;
+  }
+
+  public String getUserName() { return this.userName; }
+
+  public String getCommentBody() { return this.commentBody; }
+
+
+  @Override
+    public boolean equals(Object o) {
+        if (o == this)
+            return true;
+        if (!(o instanceof User)) {
+            return false;
+        }
+        User user = (User) o;
+        return Objects.equals(userName, user.userName) && Objects.equals(commentBody, user.commentBody);
```

Comment:
I assume you're using `Objects.equals` to handle nullability. That's totally valid. Note that there is another approach you could take, which is to require that userName and commentBody never be null (and enforcing it in the constructor). Then once the object is constructed you can always be sure the members are valid. That's pretty typical internally: we often prefer to make most members, parameters, and return values non-nullable, and mark the others with `@Nullable`. Here, I'll leave the choice up to you.